### DISCUSSION TITLE
Can someone check some statistics for c:geo?

### DISCUSSION THREAD

**Discussion by Lineflyer** [cgeo/cgeo#10001] [Q&A] [answered]
I would really like to get an impression about how many lines of codes c:geo consists of.

I found this solution to check this, which sounds rather easy (if you have a unix/linux system running, which I dont have):
https://stackoverflow.com/questions/26881441/can-you-get-the-number-of-lines-of-code-from-a-github-repository

Can someone do me/us a favor and try to get these statistics for the c:geo repo?

**Comment by UniQP** [comment, marked as answer]
Current master (df20c0d8a9d82734d307aad658069c3e9e3fb7a0):

```
cloc --exclude-dir thirdparty .
    2237 text files.
    1935 unique files.                                          
     382 files ignored.

github.com/AlDanial/cloc v 1.86  T=2.24 s (826.9 files/s, 141109.1 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
XML                            844           6904           9486         147639
Java                           725          15984          11352          77100
SVG                            151            451             96          24032
JSON                           125              0              0          22646
Gradle                           1             97            139            406
Bourne Shell                     7             55             45            193
HTML                             1              1              0             57
Markdown                         1              6              0             30
XSD                              1              0              0             17
-------------------------------------------------------------------------------
SUM:                          1856          23498          21118         272120
-------------------------------------------------------------------------------
```

**Reply by Lineflyer** [reply]
Cool thanks!

**Reply by bekuno** [reply]
> cloc --exclude-dir thirdparty .

we should also exclude the viewpagerindicator directory (in the root folder).

**Reply by UniQP** [reply]
The stats were already for the `main` directory (not clear from the provided command line, sorry), so the viewpagerindicator is already excluded. This also means that the `tests` are excluded from the statistics as well.

**Comment by moving-bits** [comment]
Codacy seems to count differently. Including third party source it sums up to 126k LOC:

![grafik](https://user-images.githubusercontent.com/3754370/107610588-410dd900-6c42-11eb-80de-3aafb4008f03.png)

See https://app.codacy.com/gh/cgeo/cgeo/dashboard?branch=master and tap on one of the date markers along the x axis.

May it be that Codacy counts multi-line commands as one, and cloc counts each line of it separately?

**Reply by Narrat** [reply]
Seems reasonable.
https://github.com/AlDanial/cloc#how-it-works-
https://github.com/AlDanial/cloc#limitations-

But as is the *nix world, there exists more than one tool for a job :D So cross checks are possible :P
https://github.com/AlDanial/cloc#other-counters-
